starter/src/main/java/com/starter/fullstack/dao/InventoryDAO.java

```diff
@@ -87,14 +91,18 @@ public Optional<Inventory> update(String id, Inventory inventory) {
    * @return Deleted Inventory.
    */
   public Optional<Inventory> delete(String id) {
-    //get Inventory by ID
-    if(retrieve(id).isPresent()){
-      Inventory ret = retrieve(id).get();
-      mongoTemplate.remove(ret);
-      return Optional.of(ret);
+    //to be used in return
+    Inventory retInventory = retrieve(id).get();
+
+    Query query = new Query(Criteria.where("id").is(id));
+    DeleteResult ret = this.mongoTemplate.remove(query, Inventory.class);
```

Comment:
With the comment I left in your last commit I mislead you with the remove method. Instead of using the remove method I meant to say try using the findAndRemove method. The code should look like the following code

`return Optional.ofNullable(this.mongoTemplate.findAndRemove(query, Inventory.class));`

This code will attempt to remove the inventory. If it doesnt then it will return and empty optional. This also removes any checks that you have to see if it was deleted or not.